test/controllers/project_skill_controller_test.exs

```diff
@@ -0,0 +1,126 @@
+    assert second_result["id"] == "#{project_skill_2.id}"
+    assert second_result["relationships"]["project"]["data"]["id"] == "#{project.id}"
+    assert second_result["relationships"]["project"]["data"]["type"] == "project"
+    assert second_result["relationships"]["skill"]["data"]["id"] == "#{phoenix.id}"
+    assert second_result["relationships"]["skill"]["data"]["type"] == "skill"
+  end
+
+  test "shows chosen resource", %{conn: conn} do
+    skill = insert(:skill)
+    project = insert(:project)
+    project_skill = insert(:project_skill, project: project, skill: skill)
+    conn = get conn, project_skill_path(conn, :show, project_skill)
+    data = json_response(conn, 200)["data"]
+    assert data["id"] == "#{project_skill.id}"
+    assert data["type"] == "project-skill"
+    assert data["attributes"] == %{}
+    assert data["relationships"]["project"]["data"]["id"] == "#{project.id}"
+    assert data["relationships"]["project"]["data"]["type"] == "project"
+    assert data["relationships"]["skill"]["data"]["id"] == "#{skill.id}"
+    assert data["relationships"]["skill"]["data"]["type"] == "skill"
+  end
+
+  test "does not show resource and instead throw error when id is nonexistent", %{conn: conn} do
+    assert_error_sent 404, fn ->
+      get conn, project_skill_path(conn, :show, -1)
+    end
+  end
+
+  test "creates and renders resource when data is valid", %{conn: conn} do
+    project = insert(:project)
+    skill = insert(:skill)
+    conn = post conn, project_skill_path(conn, :create), %{
+      "meta" => %{},
+      "data" => %{
+        "type" => "project-skill",
+        "attributes" => %{},
+        "relationships" => relationships(project, skill)
+      }
+    }
+    json = json_response(conn, 201)
+
+    id = json["data"]["id"] |> String.to_integer
+    project_skill = ProjectSkill |> Repo.get!(id)
+
+    assert json["data"]["id"] == "#{project_skill.id}"
+    assert json["data"]["type"] == "project-skill"
+    assert json["data"]["attributes"] == %{}
+    assert json["data"]["relationships"]["project"]["data"]["id"] == "#{project.id}"
+    assert json["data"]["relationships"]["project"]["data"]["type"] == "project"
+    assert json["data"]["relationships"]["skill"]["data"]["id"] == "#{skill.id}"
+    assert json["data"]["relationships"]["skill"]["data"]["type"] == "skill"
+  end
+
+  test "does not create resource and renders errors when data is invalid", %{conn: conn} do
+    conn = post conn, user_skill_path(conn, :create), %{
+      "meta" => %{},
+      "data" => %{
+        "type" => "user-skill",
+        "attributes" => %{}
+      }
+    }
+
+    assert json_response(conn, 422)["errors"] != %{}
+  end
+
+  test "deletes chosen resource", %{conn: conn} do
+    project_skill = Repo.insert! %ProjectSkill{}
+    conn = delete conn, project_skill_path(conn, :delete, project_skill)
+    assert response(conn, 204)
+    refute Repo.get(ProjectSkill, project_skill.id)
+  end
```

Comment:
Ok, so we need to add a spec for attempting to delete a non-existant record, in a style like the one at 

https://github.com/code-corps/code-corps-phoenix/blob/develop/test/controllers/project_category_controller_test.exs#L81

I've created an issue for the other ones - #151